src/iop/overexposed.c

```diff
@@ -460,8 +460,8 @@ void cleanup_global(dt_iop_module_so_t *module)
 void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pixelpipe_t *pipe,
                    dt_dev_pixelpipe_iop_t *piece)
 {
-  if(pipe->type != DT_DEV_PIXELPIPE_FULL || !self->dev->overexposed.enabled || !self->dev->gui_attached)
-    piece->enabled = 0;
+  const gboolean fullpipe = (piece->pipe->type & DT_DEV_PIXELPIPE_FULL) == DT_DEV_PIXELPIPE_FULL;
```

Comment:
Since all PIXELPIPE enums are single bits (except _ANY) this could just be
`const gboolean fullpipe = piece->pipe->type & DT_DEV_PIXELPIPE_FULL;` which is easier to read and less likely to be incorrectly partially adjusted.
I know this is spread everywhere and would not be a good search-and-replace for 4.0.1. but no need to keep spreading?

Reply:
This is not correct. It used to be but we have `DT_DEV_PIXELPIPE_FAST` adding another mask bit.

It's the exact reason why the old code was broken.

Yes - it's spread all over the place and we would either like a macro or a refactored testing function

Reply:
No - you are right :-)

Reply:
done